generator/04-syndesis-oauth-proxy.yml.mustache

```diff
@@ -0,0 +1,127 @@
+      app: syndesis
+    name: syndesis
+  spec:
+    host: ${ROUTE_HOSTNAME}
+    port:
+      targetPort: 8443
+    tls:
+      insecureEdgeTerminationPolicy: Redirect
+      termination: edge
+    to:
+      kind: Service
+      name: syndesis-oauthproxy
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    labels:
+      app: syndesis
+      component: syndesis-oauthproxy
+    name: syndesis-oauthproxy
+  spec:
+    replicas: 1
+    selector:
+      app: syndesis
+      component: syndesis-oauthproxy
+      deploymentconfig: syndesis-oauthproxy
+    strategy:
+      resources:
+        limits:
+          memory: "256Mi"
+        requests:
+          memory: "20Mi"
+      type: Recreate
+    template:
+      metadata:
+        labels:
+          app: syndesis
+          component: syndesis-oauthproxy
+          deploymentconfig: syndesis-oauthproxy
+      spec:
+        containers:
+        - name: syndesis-oauthproxy
+{{^WithDockerImages}}
+          image: ' '{{/WithDockerImages}}{{#WithDockerImages}}
+          image: ${SYNDESIS_REGISTRY}/{{Images.OAuthProxyImagePrefix}}/{{ Images.Support.OAuthProxy }}:{{ Tags.OAuthProxy }}{{/WithDockerImages}}
+          args:
+            - --http-address=:8443
+            - --https-address=:8444
+            - --provider=openshift
+            - --client-id={{#Restricted}}system:serviceaccount:${OPENSHIFT_PROJECT}:syndesis-oauth-client{{/Restricted}}{{^Restricted}}${OPENSHIFT_OAUTH_CLIENT_ID}{{/Restricted}}
+            - --client-secret=${OPENSHIFT_OAUTH_CLIENT_SECRET}
+            - --upstream=http://syndesis-rest/api/
+            - --upstream=http://syndesis-rest/mapper/
+            - --upstream=http://syndesis-atlasmap/v2/atlas/
+            - --upstream=http://syndesis-ui/
+            - --tls-cert=/etc/tls/private/tls.crt
+            - --tls-key=/etc/tls/private/tls.key
+            - --cookie-secret=$(OAUTH_COOKIE_SECRET)
+            - --pass-access-token
+            - --skip-provider-button
+            - --skip-auth-regex=/api/v1/swagger.*
+            - --skip-auth-regex=/api/v1/index.html
+            - --skip-auth-regex=/api/v1/credentials/callback
+            - --skip-auth-preflight
+            {{#Restricted}}- --openshift-sar={"namespace":"${OPENSHIFT_PROJECT}","resource":"pods","verb":"get"}{{/Restricted}}
+          env:
+          - name: OAUTH_COOKIE_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: syndesis-oauth-proxy-cookie-secret
+                key: oauthCookieSecret
+          ports:
+          - containerPort: 8443
+            name: public
+            protocol: TCP
+{{^Probeless}}
+          readinessProbe:
+            tcpSocket:
+              port: 8443
+          livenessProbe:
+            tcpSocket:
+              port: 8443{{/Probeless}}
+          volumeMounts:
+          - mountPath: /etc/tls/private
+            name: syndesis-oauthproxy-tls
+          resources:
+            limits:
+              memory: 256Mi
```

Comment:
Nitpicking, but I think 200Mi is good enough here. (Remember, the limits count up against the overall 4GB limit) ;-)

Its a good idea to keep requests low, as this helps for minishift to start low. On our cluster, we always start at half of `limit` memory.